src/main.c

```diff
@@ -251,7 +252,7 @@ int main(void)
     }
 
     // Initiate an update of the firmware.
-    if (APP_ASKS_FOR_SINGLE_TAP_RESET())
+    if (APP_ASKS_FOR_SINGLE_TAP_RESET() || uf2_dfu || serial_only_dfu)
```

Comment:
spot-on both uf2 and serial_only need USB to work, it is correct to timeout if USB is not enumerated.